### ISSUE TITLE
Blog Posts block: Show “More” Button not available with Coming Soon active

### ISSUE THREAD

**Issue by kriskarkoski** [Automattic/wp-calypso#43762]
#### Steps to reproduce

0. On a site with Coming Soon mode enabled under https://wordpress.com/settings/general/
1. Starting at URL: https://wordpress.com/block-editor/page/ (or https://wordpress.com/block-editor/post/)
2. Insert the Blog Posts block
3. Under Display Settings note the Show “More” Button option is missing

#### What I expected
To be able to enable Show “More” Button regardless of the coming soon status.

#### What happened instead
The site had to be public for the Show “More” Button option to show. 

#### Browser / OS version
Chrome 83.0.4103.116, Mac

#### Screenshot / Video
With Coming Soon on

<img width="274" alt="Screen Shot 2020-06-29 at 12 15 16 PM" src="https://user-images.githubusercontent.com/1900444/86030526-99b99a00-ba02-11ea-87af-e8dfc647c125.png">

With Coming Soon off (public site)

<img width="274" alt="Screen Shot 2020-06-29 at 12 14 42 PM" src="https://user-images.githubusercontent.com/1900444/86030570-a63df280-ba02-11ea-8947-9bfd5f4c4095.png">

#### Context / Source

#manual-testing, #user-report

**Comment by galakhyati** [comment]
Another instance here: 24122301-hc

**Comment by retnonindya** [comment]
Same issue here: 3324065-zen

**Comment by worldomonation** [comment]
Hi! I'm performing triage on older bugs as part of Quality Squad's efforts to cut down on backlog issue counts.

With the following versions:
Gutenberg: v10.2.1
CoBlocks: 2.5.3
Editing Toolkit: 3.2798

I took the following steps to reproduce:
1. create a site that is Coming Soon.
2. create a new post.
3. insert the Blog Posts block type.
4. click on the block.
5. under the Block settings, scroll under Display Settings.

I was able to observe and interact with the toggle for the Show More Posts button. 

However, when the post is previewed the `Show more` button is not present. This indicates that while the issue as originally described is no longer reproducible, the issue simply has moved down a step.
![Screen Shot 2021-03-22 at 4 03 26 PM](https://user-images.githubusercontent.com/6549265/112068996-3f301300-8b28-11eb-8ffe-b6482e508ece.png)
